dns/dns.go

```diff
@@ -0,0 +1,124 @@
+package dns
+
+import (
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/giantswarm/exporterkit"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	Namespace = "dns"
+)
+
+type Config struct {
+	Logger micrologger.Logger
+
+	Hosts []string
+}
+
+type DNSCollector struct {
+	logger micrologger.Logger
+
+	hosts []string
+
+	count      map[string]float64
+	errorCount map[string]float64
+
+	countMutex      sync.Mutex
+	errorCountMutex sync.Mutex
+
+	total      *prometheus.Desc
+	errorTotal *prometheus.Desc
+	latency    *prometheus.Desc
```

Comment:
What is the logic with these blocks here? In other places it is dependencies, internals, settings. And then alphabetically.